### ISSUE TITLE
Running "update warp" button doesn't work

### ISSUE THREAD

**Issue by superm1** [warpdotdev/Warp#5547]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

I use CachyOS and running the update warp button doesn't work.

![image](https://github.com/user-attachments/assets/1ad210b4-fa93-4bfb-bb59-e4c7b0102edf)

I notice that once I press enter the first sudo prompt is skipped without me doing anything.
![image](https://github.com/user-attachments/assets/43fd6371-ccee-4374-9786-ff898d97f973)

I then looked to see if `/etc/pacman.conf` was updated, and it wasn't.

![image](https://github.com/user-attachments/assets/89fe8866-f2a3-492d-a55d-bed4a1c5d1fb)

### To reproduce

Install old warp version.
Try to update.

### Expected behavior

It updates

### Operating system

Linux

### Operating system and version

CachyOS

### Shell Version

5.2.37(1)-release

### Current Warp version

v0.2024.11.19.08.02.stable_03

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Additional context

https://github.com/warpdotdev/Warp/issues/5544 looks similar, please mark as a duplicate if you think it is.

### Does this block you from using Warp daily?

No

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by vorporeal** [comment]
Hey @superm1; sorry about this!

First, can you confirm whether or not you're using fish as your shell?  The command shown in your screenshot looks like it's using fish syntax, but the shell version you listed uses bash's versioning scheme.  I want to make sure the issue isn't that we're suggesting the wrong command for the shell that you're using.

**Comment by superm1** [comment]
No worries!

I normally use fish, but when I pressed the button it seems like it opened a bash shell to do the update?

**Comment by vorporeal** [comment]
Interesting.  Mind deleting the pre-filled command in the update tab and instead running `echo $SHELL` so we can see shell was actually spawned for that session?

**Comment by superm1** [comment]
Huh, it is `/bin/fish`.  I was thrown off since I didn't see my directory prompt (but it's in `~` so it's actually accurate)

![image](https://github.com/user-attachments/assets/ba7acbeb-40e7-4423-a5cc-8bd65de45e9f)

**Comment by superm1** [comment]
And since it's `fish` indeed, here's the correct version string.
```
fish --version
fish, version 3.7.1
```

**Comment by superm1** [comment]
> Did you go through the full flow, i.e.: entering your password at the sudo prompt and letting it write out the file and run the pacman commands?

Well the first prompt attempt is "skipped", I didn't do anything. So I type in the second prompt attempt and nothing happens.

**Comment by vorporeal** [comment]
Yeah, I wonder if we're accidentally sending an extra enter keypress or something (so it thinks you tried to enter nothing as your password).  Any chance you could record and share a short video of you going through the flow (entering your password at the second prompt) so I can get a better sense of what's going on?

**Comment by superm1** [comment]
Here you go.

I pressed enter once on the prompt to check the command before running.
You'll see the enter got sent "through", and then I waited a while and typed my password for the second one.

[Screencast_20241206_120334.webm](https://github.com/user-attachments/assets/335ef6c1-6822-432c-b44d-cf7c19c85c94)

**Comment by vorporeal** [comment]
Ah, the command that's actually being run is not the same as the one that's in the input when you initially press enter.

Can you try wrapping the command we provide in `begin; ` and `; end`?  I think fish might not handle the open quote + newline the way we expect/in the same way as bash and zsh.

**Comment by superm1** [comment]
Yup that does it, works exactly as it should.
